src/Continuous/solve.jl

```diff
@@ -288,6 +288,16 @@ function _get_T(tspan::TimeInterval; check_zero::Bool=true, check_positive::Bool
     return T
 end
 
+function compute_nsteps(δ, tspan)
+    if iszero(δ)
+        T = zero(δ)
+        return T
+    end
+    # get time horizon from the time span imposing that it is of the form (0, T)
+    T = _get_T(tspan; check_zero=true, check_positive=true)
+    return NSTEPS = ceil(Int, T / δ)
```

Comment:
The function should always return an `Int` to be type stable.

```suggestion
        return 0
    end
    # get time horizon from the time span imposing that it is of the form (0, T)
    T = _get_T(tspan; check_zero=true, check_positive=true)
    return ceil(Int, T / δ)
```